pkg-manager/resolve-dependencies/src/resolveDependencies.ts

```diff
@@ -1305,14 +1306,19 @@ async function resolveDependency (
       optional: true,
     }
   }
+
+  // Normalize the `preferredVersion` (singular) and `preferredVersions`
+  // (plural) options. If the singular option is passed through, it'll be used
+  // instead of the plural option.
+  const preferredVersions = !options.updateRequested && options.preferredVersion != null
+    ? getExactSinglePreferredVersions(wantedDependency, options.preferredVersion)
+    : options.preferredVersions
```

Comment:
I also considered an approach that passes through a `currentPkg` object to reuse existing versions.

Unfortunately a lot more work would be needed for this approach since `currentPkg` is expected to contain information from the `packages` section of the lockfile. Since we're installing a new dependency not present before, we can't reuse an `PackageSnapshot` entry that contained peer dependency resolution results for a different package. I considered a refactor where `currentPkg` could contain more limited information. That may be worthwhile in a different PR.

Separately, I'm exploring a change where peer dependencies are "shared" for the catalog, but need to look into that further independently to see how complicated that change would be.